test/common/grpc/grpc_web_filter_test.cc

```diff
@@ -29,73 +31,60 @@ const char TRAILERS[] = "\x80\x00\x00\x00\x20grpc-status:0\r\ngrpc-message:ok\r\
-  EXPECT_EQ(std::string(MESSAGE, MESSAGE_SIZE), TestUtility::bufferToString(request_buffer));
+  bool isTextRequest() const {
+    return request_content_type() == Http::Headers::get().ContentTypeValues.GrpcWebText ||
+           request_content_type() == Http::Headers::get().ContentTypeValues.GrpcWebTextProto;
+  }
 
-  // Tests request trailers, they are passed through.
-  Http::TestHeaderMapImpl request_trailers;
-  request_trailers.addViaCopy(Http::Headers::get().GrpcStatus, "0");
-  request_trailers.addViaCopy(Http::Headers::get().GrpcMessage, "ok");
-  EXPECT_EQ(Http::FilterTrailersStatus::Continue, filter_.decodeTrailers(request_trailers));
-  EXPECT_STREQ("0", request_trailers.GrpcStatus()->value().c_str());
-  EXPECT_STREQ("ok", request_trailers.GrpcMessage()->value().c_str());
+  bool isBinaryRequest() const {
+    return request_content_type() == Http::Headers::get().ContentTypeValues.GrpcWeb ||
+           request_content_type() == Http::Headers::get().ContentTypeValues.GrpcWebProto;
+  }
 
-  // Tests response headers.
-  Http::TestHeaderMapImpl response_headers;
-  response_headers.addViaCopy(Http::Headers::get().Status, "200");
-  EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter_.encodeHeaders(response_headers, false));
-  EXPECT_EQ("200", response_headers.get_(Http::Headers::get().Status.get()));
-  EXPECT_EQ(Http::Headers::get().ContentTypeValues.GrpcWeb,
-            response_headers.ContentType()->value().c_str());
+  bool accept_text_response() {
+    return request_accept() == Http::Headers::get().ContentTypeValues.GrpcWebText ||
+           request_accept() == Http::Headers::get().ContentTypeValues.GrpcWebTextProto;
+  }
 
-  // Tests response data.
-  Buffer::OwnedImpl response_buffer(MESSAGE, MESSAGE_SIZE);
-  EXPECT_EQ(Http::FilterDataStatus::Continue, filter_.encodeData(response_buffer, false));
-  EXPECT_EQ(std::string(MESSAGE, MESSAGE_SIZE), TestUtility::bufferToString(response_buffer));
-  response_buffer.drain(response_buffer.length());
+  bool accept_binary_response() {
+    return request_accept() == Http::Headers::get().ContentTypeValues.GrpcWeb ||
+           request_accept() == Http::Headers::get().ContentTypeValues.GrpcWebProto;
+  }
 
-  // Tests response trailers.
-  Buffer::OwnedImpl trailers_buffer;
-  EXPECT_CALL(encoder_callbacks_, addEncodedData(_))
-      .WillOnce(Invoke([&](Buffer::Instance& data) { trailers_buffer.move(data); }));
-  Http::TestHeaderMapImpl response_trailers;
-  response_trailers.addViaCopy(Http::Headers::get().GrpcStatus, "0");
-  response_trailers.addViaCopy(Http::Headers::get().GrpcMessage, "ok");
-  EXPECT_EQ(Http::FilterTrailersStatus::Continue, filter_.encodeTrailers(response_trailers));
-  EXPECT_EQ(std::string(TRAILERS, TRAILERS_SIZE), TestUtility::bufferToString(trailers_buffer));
+  GrpcWebFilter filter_;
+  NiceMock<Http::MockStreamEncoderFilterCallbacks> encoder_callbacks_;
+};
+
+TEST_F(GrpcWebFilterTest, SupportedContentTypes) {
```

Comment:
How come this is under `GrpcWebFilterTest`? It will run a combinatorial number of times but isn't itself parameterized.